### PR TITLE
Improve docs in `sighash` and `psbt/mod.rs`

### PR DESCRIPTION
**Pull request by tcharding**
Done while working on sighash and PSBT signing. Just the usual docs fixes. Note, does not do the whole `psbt` module just the file mentioned.

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK 52ec1cc56b10c9402ee1341fff0a212660033479

**Comment by tcharding** [comment]
Converting to draft while we do the MSRV release.

**Comment by tcharding** [comment]
Rebased after the edition 2018 shenanigans.

**Review by apoelstra — APPROVED** [review]
ACK 9896f27eae786329967328c487db5e09a049d973

File: src/util/psbt/mod.rs [hunk 1]
```diff
@@ -119,8 +115,11 @@ impl PartiallySignedTransaction {
...
     }
 
-    /// Create a PartiallySignedTransaction from an unsigned transaction, error
-    /// if not unsigned
+    /// Creates a PSBT from an unsigned transaction.
+    ///
+    /// # Errors
+    ///
+    /// If transactions is not unsigned.
     pub fn from_unsigned_tx(tx: Transaction) -> Result<Self, Error> {
```
Comment (sanket1729) [inline src/util/psbt/mod.rs:123]: Aside. As noted elsewhere, If we are consuming a value, we should return it in the `Error`. Perhaps a good first issue

**Review by sanket1729 — APPROVED** [review]
ACk 9896f27eae786329967328c487db5e09a049d973

**Pull request merged**